test/nodes/test_lost_in_the_middle.py

```diff
@@ -0,0 +1,189 @@
+        Document("word2"),
+        Document("word3"),
+        Document("word4"),
+        Document("word5"),
+        Document("word6"),
+        Document("word7"),
+        Document("word8"),
+        Document("word9"),
+    ]
+    result, _ = litm.run(query="", documents=docs)
+    expected_order = "word1 word3 word5 word6 word4 word2".split()
+    assert all(doc.content == expected_order[idx] for idx, doc in enumerate(result["documents"]))
+
+    litm = LostInTheMiddleRanker(word_count_threshold=9)
+    result, _ = litm.run(query="", documents=docs)
+    expected_order = "word1 word3 word5 word7 word9 word8 word6 word4 word2".split()
+    assert all(doc.content == expected_order[idx] for idx, doc in enumerate(result["documents"]))
+
+
+@pytest.mark.unit
+def test_word_count_threshold_greater_than_total_number_of_words_returns_all_documents():
+    ranker = LostInTheMiddleRanker(word_count_threshold=100)
+    docs = [
+        Document("word1"),
+        Document("word2"),
+        Document("word3"),
+        Document("word4"),
+        Document("word5"),
+        Document("word6"),
+        Document("word7"),
+        Document("word8"),
+        Document("word9"),
+    ]
+    ordered_docs = ranker.predict(query="test", documents=docs)
+    assert len(ordered_docs) == len(docs)
+    expected_order = "word1 word3 word5 word7 word9 word8 word6 word4 word2".split()
+    assert all(doc.content == expected_order[idx] for idx, doc in enumerate(ordered_docs))
+
+
+@pytest.mark.unit
+def test_truncation_with_threshold():
+    # tests that truncation works as expected
+    litm = LostInTheMiddleRanker(word_count_threshold=9, truncate_document=True)
+    docs = [
+        Document("word1 word1"),
+        Document("word2 word2"),
+        Document("word3 word3"),
+        Document("word4 word4"),
+        Document("word5 word5"),
+        Document("word6 word6"),
+        Document("word7 word7"),
+        Document("word8 word8"),
+        Document("word9 word9"),
+    ]
+    result, _ = litm.run(query="", documents=docs)
+    expected_order = "word1 word1 word3 word3 word5 word4 word4 word2 word2"
+    assert expected_order == " ".join(doc.content for doc in result["documents"])
+
+
+@pytest.mark.unit
+def test_empty_documents_returns_empty_list():
+    ranker = LostInTheMiddleRanker()
+    assert ranker.predict(query="test", documents=[]) == []
+
+
+@pytest.mark.unit
+def test_list_of_one_document_returns_same_document():
+    ranker = LostInTheMiddleRanker()
+    doc = Document(content="test", content_type="text")
+    assert ranker.predict(query="test", documents=[doc]) == [doc]
+
+
+@pytest.mark.unit
+def test_non_textual_documents():
+    #  tests that merging a list of non-textual documents raises a ValueError
+    litm = LostInTheMiddleRanker()
+    doc1 = Document(content="This is a textual document.")
+    doc2 = Document(content_type="image", content="This is a non-textual document.")
+    with pytest.raises(ValueError):
```

Comment:
you can add `match="provided documents are not textual"` here in `pytest.raises(...)`